packages/data/src/components/with-select/test/index.js

```diff
@@ -190,50 +173,45 @@ describe( 'withSelect', () => {
 
 		const renderSpy = jest.spyOn( OriginalComponent.prototype, 'render' );
 
-		const mapSelectToProps = jest
-			.fn()
-			.mockImplementation( ( _select ) => ( {
-				count: _select( 'counter' ).getCount(),
-			} ) );
+		const mapSelectToProps = jest.fn( ( _select ) => ( {
+			count: _select( 'counter' ).getCount(),
+		} ) );
 
-		const mapDispatchToProps = jest
-			.fn()
-			.mockImplementation( ( _dispatch ) => ( {
-				increment: _dispatch( 'counter' ).increment,
-			} ) );
+		const mapDispatchToProps = jest.fn( ( _dispatch ) => ( {
+			increment: _dispatch( 'counter' ).increment,
+		} ) );
 
 		const DataBoundComponent = compose( [
 			withSelect( mapSelectToProps ),
 			withDispatch( mapDispatchToProps ),
 		] )( OriginalComponent );
 
-		let testRenderer, testInstance;
-		const createTestRenderer = () =>
-			TestRenderer.create(
+		it( 'should rerun if had dispatched action during mount', () => {
+			const { container, unmount } = render(
 				<RegistryProvider value={ testRegistry }>
 					<DataBoundComponent />
 				</RegistryProvider>
 			);
-		act( () => {
-			testRenderer = createTestRenderer();
-		} );
-		testInstance = testRenderer.root;
-		it( 'should rerun if had dispatched action during mount', () => {
-			expect( testInstance.findByType( 'div' ).props.children ).toBe( 2 );
+
+			expect( container ).toHaveTextContent( '2' );
 			// Expected 3 times because:
 			// - 1 on initial render
 			// - 1 on effect before subscription set.
 			// - 1 for the rerender because of the mapOutput change detected.
 			expect( mapSelectToProps ).toHaveBeenCalledTimes( 3 );
 			expect( renderSpy ).toHaveBeenCalledTimes( 2 );
+
+			unmount();
 		} );
+
 		it( 'should rerun on unmount and mount', () => {
-			act( () => {
-				testRenderer.unmount();
-				testRenderer = createTestRenderer();
-			} );
-			testInstance = testRenderer.root;
-			expect( testInstance.findByType( 'div' ).props.children ).toBe( 4 );
+			const { container } = render(
+				<RegistryProvider value={ testRegistry }>
+					<DataBoundComponent />
+				</RegistryProvider>
+			);
+
+			expect( container ).toHaveTextContent( '4' );
```

Comment:
Technically, we could use `screen.getByText()` to use a screen query instead of `container`, but I don't think there's any ESLint rule forbidding us to use just `container` without its methods, so this should be good as-is.

Reply:
Updated to use `role="status"` and `screen.getByRole()` this matches queries in `useSelect` tests - [5e1600d](https://github.com/WordPress/gutenberg/pull/45058/commits/5e1600d48556d37b9323b6b63e5d9765f82d0a27).